File: android/src/main/kotlin/com/compress/all/flutter_compress/ImageConfig.kt
package com.compress.all.flutter_compress

/** Parsed mirror of the Dart [ImageCompressConfig]. */
data class ImageConfig(
    /** Requested output format, or null to keep the source's. */
    val format: String?,
    val quality: Int,
    val targetSizeKB: Int?,
    val maxWidth: Int?,
    val maxHeight: Int?,
    val keepExif: Boolean,
    val lossless: Boolean,
    val keepOriginalIfLarger: Boolean,
    val minSavingsPercent: Int,
) {
    companion object {
        fun fromMap(m: Map<String, Any?>) = ImageConfig(
            format = m["format"] as? String,
            quality = (m["quality"] as? Number)?.toInt() ?: 85,
            targetSizeKB = (m["targetSizeKB"] as? Number)?.toInt(),
            maxWidth = (m["maxWidth"] as? Number)?.toInt(),
            maxHeight = (m["maxHeight"] as? Number)?.toInt(),
            keepExif = m["keepExif"] as? Boolean ?: false,
            lossless = m["lossless"] as? Boolean ?: false,
            keepOriginalIfLarger = m["keepOriginalIfLarger"] as? Boolean ?: true,
            minSavingsPercent = (m["minSavingsPercent"] as? Number)?.toInt() ?: 0,
        )
    }
}

File: darwin/flutter_compress_pro/Sources/flutter_compress_pro/ImageConfig.swift
import Foundation

/// Parsed mirror of the Dart `ImageCompressConfig`.
struct ImageConfig {
  /// Requested output format, or nil to keep the source's.
  let format: String?
  let quality: Int
  let targetSizeKB: Int?
  let maxWidth: Int?
  let maxHeight: Int?
  let keepExif: Bool
  let lossless: Bool
  let keepOriginalIfLarger: Bool
  let minSavingsPercent: Int

  init(map: [String: Any]) {
    format = map["format"] as? String
    quality = (map["quality"] as? NSNumber)?.intValue ?? 85
    targetSizeKB = (map["targetSizeKB"] as? NSNumber)?.intValue
    maxWidth = (map["maxWidth"] as? NSNumber)?.intValue
    maxHeight = (map["maxHeight"] as? NSNumber)?.intValue
    keepExif = map["keepExif"] as? Bool ?? false
    lossless = map["lossless"] as? Bool ?? false
    keepOriginalIfLarger = map["keepOriginalIfLarger"] as? Bool ?? true
    minSavingsPercent = (map["minSavingsPercent"] as? NSNumber)?.intValue ?? 0
  }
}
